.github/workflows/build.yml

```diff
@@ -4,9 +4,13 @@ on:
   push:
     branches: [ main ]
     tags:
-      - "*"
+      - '*'
   pull_request:
-    branches: [main, "release/*"]
+    branches: [main, 'release/*']
+
+env:
+  STABLE_TAG: ${{ github.event_name == 'push' && github.ref_name || format('pr-{0}', github.event.pull_request.number) }}
```

Comment:
"push" event is only when you push (merge) to main branch, or you push (create) tag. When you (force) push to PR branch, it generates "pull_request" event of "synchronize" type.

This effectively means:
- if you push to main branch, then use main tag
- if you create tag, then use this git tag as container image tag
- if you push to PR, then use pr-NUMBER as a tag